homeassistant/components/monarchmoney/coordinator.py

```diff
@@ -0,0 +1,168 @@
+"""Data coordinator for monarch money."""
+
+import asyncio
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Any
+
+from aiohttp import ClientResponseError
+from gql.transport.exceptions import TransportServerError
+from monarchmoney import LoginFailedException, MonarchMoney
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import LOGGER
+
+
+@dataclass
+class MonarchAccount:
+    """Dataclass to store & parse account data from monarch accounts."""
+
+    id: str
+    logo_url: str | None
+    name: str
+    balance: float
+    type: str  # type will be used for icons
+    type_name: str  # type name will be used for device
+    subtype: str
+    subtype_name: str
+    data_provider: str
+    institution_url: str | None
+    institution_name: str | None
+    last_update: datetime
+    date_created: datetime
+
+    @property
+    def is_value_account(self):
+        """Return true if we are tracking a value type asset."""
+        return self.type in ["real-estate", "vehicle", "valuables", "other_assets"]
```

Comment:
I like all these dataclasses, but why don't we move this to the library?

Reply:
If you then add a `py.typed` file and make sure your types are good, you have an amazing library

Reply:
> I like all these dataclasses, but why don't we move this to the library?

I don't maintain the library... its 3rd party